packages/grid/veilid/server/veilid_streamer.py

```diff
@@ -0,0 +1,441 @@
+# stdlib
+import asyncio
+from collections.abc import Callable
+from collections.abc import Coroutine
+from enum import ReprEnum
+from enum import nonmember
+import hashlib
+import logging
+import math
+from struct import Struct
+from typing import Any
+import uuid
+
+# third party
+import veilid
+
+# relative
+from .constants import MAX_MESSAGE_SIZE
+from .utils import retry
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+# An asynchronous callable type hint that takes bytes as input and returns bytes
+AsyncReceiveStreamCallback = Callable[[bytes], Coroutine[Any, Any, bytes]]
+CallId = bytes
+
+
+class RequestType(bytes, ReprEnum):
+    SIZE = nonmember(8)
+
+    STREAM_START = b"@VS@SS"
+    STREAM_CHUNK = b"@VS@SC"
+    STREAM_END = b"@VS@SE"
+    STREAM_SINGLE = b"@VS@S1"  # Special case for handling single chunk messages
+
+    def __init__(self, value: bytes) -> None:
+        # Members must be a bytes object of length == SIZE. If length is less than
+        # SIZE, it'll be padded with null bytes to make it SIZE bytes long. If it is
+        # greater, a ValueError will be raise.
+        size = int(self.SIZE)  # type: ignore
+        if len(value) > size:
+            raise ValueError("Value must not be greater than 8 in length")
+        if len(value) < size:
+            value = value.ljust(size, b"\x00")
+        self._value_ = value
+
+    def __eq__(self, __other: object) -> bool:
+        return self._value_ == __other
+
+
+class ResponseType(bytes, ReprEnum):
+    OK = b"@VS@OK"
+    ERROR = b"@VS@ER"
```

Comment:
MAYBE:  we could keep the bytes value: b"@VS@ERR"

Reply:
I wanted to keep the length same for both the `ResponseType`s